### PR TITLE
docs: add doc about custom resources and parameters

### LINKED ISSUE #3039: Overwrite & customize resolvers Overwriting Resolvers documentation usage question
**Which Category is your question related to?**
Related to this [issue](https://github.com/aws-amplify/amplify-ios/issues/1036)
A customer reported an issue which is to add a custom resolver following [CLI doc](https://docs.amplify.aws/cli/graphql-transformer/resolvers#add-a-custom-geolocation-search-resolver-that-targets-an-elasticsearch-domain-created-by-searchable), but I failed to reproduce it. 

I think it's better to improve the documentation so that it is easier to understand and follow, say:
- how to create the resolver record in the stack’s Resources block?
- which field I should replace with in the stack resource?
- what's the ApiId field in the stack resource example? I tried to replace it with the `api id` I found from my AppSync console, but still failed to do the `amplify push`
```
"ApiId": {
            "Ref": "AppSyncApiId"
        },
```
etc.

**Amplify CLI Version** 
4.41.2

**What AWS Services are you utilizing?**
Amplify.API or AWS AppSync

### PR DESCRIPTION
**Pull request by yuth**
fix #3039

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/cli/graphql-transformer/resolvers.md [hunk 1]
```diff
@@ -18,6 +18,93 @@ and you want to change the behavior of request mapping template for the *Query.g
 
 ## Custom Resolvers
 
+GraphQL transformer supports adding custom `Query`, `Mutation` and `Subscription` when the generated ones do not cover your use-case. This is achieved by adding the required `Query`, `Mutation` or `Subscription` type to your schema and then adding resolvers to newly added field by creating a custom stack inside `<project-root>/amplify/backend/api/<api-name>/stacks` directory of your API. To add the custom fields add the following to your schema
```
Comment (renebrandel) [inline docs/cli/graphql-transformer/resolvers.md:21]:
```suggestion
You can add custom `Query`, `Mutation` and `Subscription` when the generated ones do not cover your use case. To do this, first add the required `Query`, `Mutation` or `Subscription` type to your schema. Then, add resolvers for the newly added fields by creating a custom stack inside `<project-root>/amplify/backend/api/<api-name>/stacks` directory of your API.

To add the custom fields, add the following to your schema:
```

File: docs/cli/graphql-transformer/resolvers.md [hunk 2]
```diff
@@ -18,6 +18,93 @@ and you want to change the behavior of request mapping template for the *Query.g
...
+    # Add all the custom mutations here
+  }
+
+  type Subscription {
+    # Add all the custom subscription here
+  }
+
+```
+
+Transformer by default creates a file called `CustomResources.json` inside `<project-root>/amplify/backend/api/<api-name>/stacks`, which can be used to add the custom resolvers for newly added `Query`, `Mutation` or `Subscription`. The custom stack gets the following arguments passed to it, allowing you to get details about API
```
Comment (renebrandel) [inline docs/cli/graphql-transformer/resolvers.md:40]:
```suggestion
The GraphQL Transformer by default creates a file called `CustomResources.json` inside `<project-root>/amplify/backend/api/<api-name>/stacks`, which can be used to add the custom resolvers for newly added `Query`, `Mutation` or `Subscription`. The custom stack gets the following arguments passed to it, allowing you to get details about API:
```

File: docs/cli/graphql-transformer/resolvers.md [hunk 3]
```diff
@@ -18,6 +18,93 @@ and you want to change the behavior of request mapping template for the *Query.g
...
+                        }
+                    ]
+                }
+            }
+        }
+    }
+}
+```
+
+The request and response template should be placed inside `<project-root>/amplify/backend/api/<api-name>/resolvers` folder. Resolver templates are written using VTL. For more details refer to [AWS AppSync VTL documentation](https://docs.aws.amazon.com/appsync/latest/devguide/resolver-mapping-template-reference.html)
```
Comment (renebrandel) [inline docs/cli/graphql-transformer/resolvers.md:106]: what about the resolver file name? We should add a note on its format.

Comment (yuth) [reply docs/cli/graphql-transformer/resolvers.md:106]: When these are custom the customer can name it however they see it fit. They should use the same name in the `CustomResources.json`

File: docs/cli/graphql-transformer/resolvers.md [hunk 4]
```diff
@@ -18,6 +18,95 @@ and you want to change the behavior of request mapping template for the *Query.g
 
 ## Custom Resolvers
 
+You can add custom `Query`, `Mutation` and `Subscription` when the generated ones do not cover your use case. To do this, first add the required `Query`, `Mutation` or `Subscription` type to your schema. Then, add resolvers for the newly added fields by creating a custom stack inside `<project-root>/amplify/backend/api/<api-name>/stacks` directory of your API.
```
Comment (renebrandel) [inline docs/cli/graphql-transformer/resolvers.md:21]: How about this:
```suggestion
You can add custom `Query`, `Mutation` and `Subscription` when the generated ones do not cover your use case. 

1. Add the required `Query`, `Mutation` or `Subscription` type to your schema. 
2. Add resolvers for the newly added fields by creating a custom stack inside `<project-root>/amplify/backend/api/<api-name>/stacks` directory of your API
3. Edit the resolver ETC ETC ETC ETC
```

File: docs/cli/graphql-transformer/resolvers.md (hunk 3 above)
Comment (renebrandel) [reply docs/cli/graphql-transformer/resolvers.md:108]: I'd recommend putting a default here that we recommend.

**Review by renebrandel — APPROVED** [review]

**Pull request merged**